File: printafter.pl
#!/usr/bin/env perl
use strict;
use warnings;

my $filename = $ARGV[0];
my $date = $ARGV[1];
open my $fh, $filename or die "Couldn't open $filename: $!";
my $print = 0;
while( my $line = <$fh> ){
    if (!$print && $line =~ m/^$date/) {
        $print = 1;
    }
    print $line if ($print == 1);
}

File: printafter.py
#!/usr/bin/env python3
import sys
import re


if __name__ == '__main__':
    fn = sys.argv[1]
    date1 = sys.argv[2]
    date2 = sys.argv[3]
    printing = False
    for line in open(fn, 'r'):
        if date1 in line:
            printing = True
        if printing:
            print(line)
